src/date.rs

```diff
@@ -206,9 +206,13 @@ impl Date {
     /// assert_eq!(d.to_string(), "2022-06-07");
     /// ```
     pub fn from_timestamp(timestamp: i64, require_exact: bool) -> Result<Self, ParseError> {
-        let (timestamp_second, _) = Self::timestamp_watershed(timestamp)?;
+        let (timestamp_second, millis) = Self::timestamp_watershed(timestamp)?;
         let d = Self::from_timestamp_calc(timestamp_second)?;
         if require_exact {
+            if millis != 0 {
+                return Err(ParseError::DateNotExact);
+            }
+
```

Comment:
Why do we need to raise an error here?

Reply:
Also, isn't the second returned value in `microseconds`?

Reply:
I guess maybe I don't understand the date not exact concept...

Reply:
Yes, it's in microseconds. I'll split into a separate PR as I think this is a correctness fix but sort of drive-by here.

Reply:
I don't mind the fix being here, but was just confused about the var name here

Reply:
#67